systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterTest.java

```diff
@@ -0,0 +1,216 @@
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.strimzi.test.ClusterController;
+import io.strimzi.test.KafkaCluster;
+import io.strimzi.test.Namespace;
+import io.strimzi.test.OpenShiftOnly;
+import io.strimzi.test.Resources;
+import io.strimzi.test.StrimziRunner;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.k8s.KubeClient;
+import io.strimzi.test.k8s.KubeClusterException;
+import io.strimzi.test.k8s.KubeClusterResource;
+import io.strimzi.test.k8s.Oc;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import static io.strimzi.test.TestUtils.map;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(StrimziRunner.class)
+@Namespace(KafkaClusterTest.NAMESPACE)
+@ClusterController
+public class KafkaClusterTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaClusterTest.class);
+
+    public static final String NAMESPACE = "kafka-cluster-test";
+
+    @ClassRule
+    public static KubeClusterResource cluster = new KubeClusterResource();
+
+    private KubeClient<?> kubeClient = cluster.client();
+
+    private static String kafkaStatefulSetName(String clusterName) {
+        return clusterName + "-kafka";
+    }
+
+    private static String kafkaPodName(String clusterName, int podId) {
+        return kafkaStatefulSetName(clusterName) + "-" + podId;
+    }
+
+    private static String zookeeperStatefulSetName(String clusterName) {
+        return clusterName + "-zookeeper";
+    }
+
+    private static String zookeeperPodName(String clusterName, int podId) {
+        return zookeeperStatefulSetName(clusterName) + "-" + podId;
+    }
+
+    @BeforeClass
+    public static void waitForCc() {
+        // TODO Build this into the annos, or get rid of the annos
+        //cluster.client().waitForDeployment("strimzi-cluster-controller");
+    }
+
+    @Test
+    @Resources(value = "../examples/templates/cluster-controller", asAdmin = true)
+    @OpenShiftOnly
+    public void testDeployKafkaClusterViaTemplate() {
+        Oc oc = (Oc) this.kubeClient;
+        String clusterName = "openshift-my-cluster";
+        oc.newApp("strimzi-ephemeral", map("CLUSTER_NAME", clusterName));
+        oc.waitForStatefulSet(zookeeperStatefulSetName(clusterName), 1);
+        oc.waitForStatefulSet(kafkaStatefulSetName(clusterName), 3);
+        oc.deleteByName("cm", clusterName);
+        oc.waitForResourceDeletion("statefulset", kafkaStatefulSetName(clusterName));
+        oc.waitForResourceDeletion("statefulset", zookeeperStatefulSetName(clusterName));
+    }
+
+    private void replaceCm(String cmName, String fieldName, String fieldValue) {
```

Comment:
Looks like we can move this method to some util or abstract or base class because I think we will replace CM in feature many times in future. So this method can be used in several test classes. WDYT?

Reply:
I know it's tempting, but I usually try to resist the urge to do things like that until I actually need to use it elsewhere, and then factor it out. Otherwise you can easily end up with lots of "utility" methods, which you thought were useful, but are actually only used in one place. It is always a bit of a judgement call though.

Reply:
Absolutely agree